frontend/app-development/layout/version-control/VersionControlHeader.tsx

```diff
@@ -157,22 +160,27 @@ export const VersionControlHeader = (props: IVersionControlHeaderProps) => {
       });
     }
   };
+
   const repoPushMutation = useRepoPushMutation(org, app);
   const pushChanges = async () => {
+    setAltinnSpinnerState(true);
     setModalState({
       ...initialModalState,
       header: t('sync_header.sharing_changes'),
       isLoading: true,
     });
-    await repoPushMutation.mutate();
+    repoPushMutation.mutate();
     setHasChangesInMaster(false);
     setHasChangesInLocalRepo(false);
+    await new Promise((resolve) => setTimeout(resolve, 1000));
```

Comment:
I have tested this and it does appear to work, but it doesn't solve the actual problem, which is that we should wait for the result before showing the success-message. If the call takes more than 1s, we might still run into the same issue.
Looks like we should have been using the `mutateAsync` method rather than `mutate` here. See suggestion above 😊 

The `useRepoPushMutation` hook has an `onSuccess` function where we could set the spinner state, by passing the function that sets spinner state, see suggestions above. Let me know if you have any questions!